### PR TITLE
LossMisconfigurationError

### PR DESCRIPTION
**Pull request by piraka9011**
# What does this PR do ?

Add a LossMisconfigurationException that provides more context on how to fix certain issues that can occur during training.

**Collection**: ASR

# Changelog 

- Add a `LossMisconfigurationException` for `ContrastiveLoss`

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [X] New Feature
- [ ] Bugfix
- [X] Documentation

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to #4587

### CODE REVIEW

File: nemo/utils/exceptions.py [hunk 1]
```diff
@@ -35,3 +35,13 @@ def __call__(self, *args, **kwargs):
...
     def __getattr__(self, *args, **kwargs):
         raise LightningNotInstalledException(self)
+
+
+class LossMisconfigurationException(NeMoBaseException):
+    def __init__(self, obj):
+        message = (
+            f"Error while computing the loss in {obj}. Make sure your loss parameters are configured "
+            "properly to match your dataset setup (ex. making sure `train_ds.min_duration` is long "
+            "enough for `ContrastiveLoss.num_negatives`.)"
```
Comment (sam1373) [inline nemo/utils/exceptions.py:45]: Perhaps move this text to only the contrastive loss instance of this exception?

Comment (piraka9011) [reply nemo/utils/exceptions.py:45]: Sure, can add a separate `msg` arg where someone can provide a custom message

Comment (jbalam-nv) [reply nemo/utils/exceptions.py:45]: @piraka9011 Could you please update the PR with the change requested above? Thanks!

Comment (piraka9011) [reply nemo/utils/exceptions.py]: Sorry! Missed this. See latest changes.

**Pull request closed without merging**